driver-core/src/main/java/com/datastax/driver/core/policies/WhiteListPolicy.java

```diff
@@ -61,4 +74,51 @@ public boolean apply(Host host) {
       }
     };
   }
+
+  /**
+   * Creates a new policy with the given host names.
+   *
+   * <p>See {@link #fromHosts(LoadBalancingPolicy, String...)} for more details.
+   */
+  public static WhiteListPolicy fromHosts(
+      LoadBalancingPolicy childPolicy, Collection<String> hostnames) {
+    return fromHosts(childPolicy, hostnames.toArray(new String[0]));
+  }
+
+  /**
+   * Creates a new policy that wraps the provided child policy but only "allows" hosts having
+   * addresses that match those from the resolved input host names.
+   *
+   * <p>Note that all host names must be resolvable; if <em>any</em> of them cannot be resolved,
+   * this method will fail.
+   *
+   * @param childPolicy the wrapped policy.
+   * @param hostnames list of host names to resolve white listed addresses from.
+   * @throws IllegalArgumentException if any of the given {@code hostnames} could not be resolved.
+   * @throws NullPointerException If null was provided for a hostname.
+   * @throws SecurityException if a security manager is present and permission to resolve the host
+   *     name is denied.
+   */
+  public static WhiteListPolicy fromHosts(LoadBalancingPolicy childPolicy, String... hostnames) {
+    final Set<InetAddress> validAddresses = new HashSet<InetAddress>();
```

Comment:
Shouldn't we use a concurrent/immutable structure here, e.g. `ImmutableSet`? It will be accessed concurrently from within the predicate created below so I think it would be safer to make it thread-safe and immutable.

Reply:
Makes sense to do that, especially to be consistent with the other predicate implementation.